### PR TITLE
Automated cherry pick of #67097: Ignore EIO error in unmount path

### PR DESCRIPTION
**Pull request by chakri-nelluri**
Cherry pick of #67097 on release-1.10.

#67097: Ignore EIO error in unmount path

### CODE REVIEW

**Comment by chakri-nelluri** [comment]
```release-note
NONE
```

**Comment by chakri-nelluri** [comment]
@saad-ali @gnufied @msau42 Please approve it for cherry-pick

File: pkg/volume/flexvolume/detacher.go [hunk 1]
```diff
@@ -49,17 +49,24 @@ func (d *flexVolumeDetacher) Detach(volumeName string, hostName types.NodeName)
...
 	}
+	if pathErr != nil && !util.IsCorruptedMnt(pathErr) {
+		return fmt.Errorf("Error checking path: %v", pathErr)
+	}
 
 	notmnt, err := isNotMounted(d.plugin.host.GetMounter(d.plugin.GetPluginName()), deviceMountPath)
 	if err != nil {
-		return err
+		if util.IsCorruptedMnt(err) {
+			notmnt = false // Corrupted error is assumed to be mounted.
```
Comment (rata) [inline pkg/volume/flexvolume/detacher.go]: Just curious: not sure why is assumed to be mounted if the error can be ENOTCONN

But this is identical to the original, so the same should be here in the cherry-pick :)

**Review by rata — APPROVED** [review]

**Comment by MaciekPytel** [comment]
Can you add release note to #67097? Ideally release notes for patch release should mention every non-trivial change in the release, so the users know what to expect when they upgrade.

Release notes are generated from original PRs to master, not cherry-picks, so adding the note just to this PR won't make it show up in release notes.

**Comment by MaciekPytel** [comment]
Also: I've approved cherry-pick, but this needs to be approved by an owner of pkg/volume/util today to make it to 1.10.7.

**Pull request merged**